tools/GenerateCompilationDatabase.ps1

```diff
@@ -0,0 +1,166 @@
+Param(
+    [string]$Ms2ccVersion = "1.3.0",
+    [switch]$Update
+
+)
+
+$binlogFileBase = Join-Path (Split-Path $PSScriptRoot -parent) "BuildOutput\Binlogs"
+$binlogFiles = Get-ChildItem -Path $binlogFileBase -Filter *.binlog -Recurse | Select-Object -ExpandProperty FullName
+
+Write-Host "Binlog files found:"
+$binlogFiles | ForEach-Object { Write-Host $_ }
+
+$VCToolsInstallDir = . "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe" -Latest -prerelease -requires Microsoft.Component.MSBuild -property InstallationPath
+write-host "VCToolsInstallDir: $VCToolsInstallDir"
+
+$msBuildPath = "$VCToolsInstallDir\MSBuild\Current\Bin\msbuild.exe"
+write-host "msBuildPath: $msBuildPath"
+
+Remove-Item "temp-filtered.log" -ErrorAction SilentlyContinue
+Remove-Item "temp-filtered2.log" -ErrorAction SilentlyContinue
+
+# Target "ClCompile" in file "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets" from project "C:\WindowsAppSDK\dev\Detours\Detours.vcxproj" (target "_ClCompile" depends on it):
+# Context here would be: "C:\WindowsAppSDK\dev\Detours"
+
+foreach ($binlogFile in $binlogFiles) {
+    if (-Not (Test-Path $binlogFile)) {
+        Write-Error "Binlog file not found: $binlogFile"
+        exit 1
+    }
+
+    & $msBuildPath $binlogFile /v:normal /noconlog /flp:logfile=temp.log
```

Comment:
Where's that writing temp.log? If current directory please switch to `Join-Path (Get-ProjectRoot) "temp"` for the per-repo temp location (created by DevCheck)